examples/2d/elixir_lbm_couette.jl

```diff
@@ -0,0 +1,81 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the Lattice-Boltzmann equations for the D2Q9 scheme
+
+equations = LatticeBoltzmannEquations2D(Ma=0.05, Re=2000)
+
+initial_condition = initial_condition_couette_unsteady
+boundary_conditions = (
+                       x_neg=boundary_condition_periodic,
+                       x_pos=boundary_condition_periodic,
+                       y_neg=boundary_condition_wall_noslip,
+                       y_pos=boundary_condition_couette,
+                      )
+
+surface_flux = flux_lax_friedrichs
+solver = DGSEM(3, surface_flux)
+
+coordinates_min = (0, 0)
+coordinates_max = (1, 1)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=3,
+                periodicity=(true, false),
+                n_cells_max=10_000,)
+
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
+                                    boundary_conditions=boundary_conditions)
+
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 40.0)
+ode = semidiscretize(semi, tspan)
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 1000
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+save_restart = SaveRestartCallback(interval=1000,
+                                   save_final_restart=true)
+
+# Custom solution variables: normalize velocities by reference speed `u0`
+@inline function macroscopic_normalized(u, equations)
+  macroscopic = cons2macroscopic(u, equations)
+  rho, v1, v2, p = macroscopic
+
+  # Use `typeof(macroscopic)` to avoid having to explicitly add `using StaticArrays`
+  typeof(macroscopic)(rho, v1/equations.u0, v2/equations.u0, p)
+end
+Trixi.varnames(::typeof(macroscopic_normalized), equations) = ("rho", "v1_normalized", "v2_normalized", "p")
```

Comment:
@ranocha Could you please check if this is kosher? Especially the use of `typeof(macroscopic)` to avoid `using StaticArrays`?

Reply:
I think it would be a bit better to write this as
```julia
convert(typeof(macroscopic), (rho, v1/equations.u0, v2/equations.u0, p))
```

Reply:
Great, thanks a lot for checking and improving this!